### PR TITLE
[Parse] Avoid 'inout' operator passing in Character.Info testing

### PR DESCRIPTION
**Pull request by rintaro**
Passing `var` value with `&` operator triggers TSAN "modifying access" detection even though it doesn't actually mutates the value. That results "access race" sanitizer error in multi-thread environments.

Instead of using a tuple table, use 'switch' to test the character info. Also, eliminate unused character info value, and introduce 'IDENT_CONT'

https://github.com/apple/swift-syntax/issues/893 
rdar://100906576

### CODE REVIEW

**Comment by rintaro** [comment]
@swift-ci Please test

File: Sources/SwiftParser/CharacterInfo.swift [hunk 1]
```diff
@@ -42,167 +34,70 @@ extension Unicode.Scalar {
   /// to be allowed to appear in a starting position in a programming language
   /// identifier.
   var isAsciiIdentifierStart: Bool {
-    return self.testRawInfoTable { entry in
-      !entry.intersection([ .UPPER, .LOWER, .UNDER, ]).isEmpty
-    }
+    self.testCharacterInfo(.IDENT_CONT) && !self.isDigit && self != "$"
```
Comment (CodaFi) [inline Sources/SwiftParser/CharacterInfo.swift:37]: Maybe we should burn a bit for identifier starts too.

**Review by CodaFi — APPROVED** [review]
It's great to see this table continue to shrink to only the relevant bits.

**Comment by rintaro** [comment]
@swift-ci Please test

**Pull request merged**